<launch>
    <arg name="model" default="$(find hak_package)/urdf/humanArm5.urdf"/>
    <arg name="gui" default="false"/>
    <arg name="rvizconfig" default="$(find hak_package)/rviz/urdf.rviz" />
    <arg name="s1_name" default="LPMSRS2000176"/>
    <arg name="s2_name" default="LPMSRS2000177"/>
    <arg name="s1_node_name" default="s1_node"/>
    <arg name="s2_node_name" default="s2_node"/>
    <arg name="s1_topic_name" default="/s1_imu"/>
    <arg name="s2_topic_name" default="/s2_imu"/>

    <param name="robot_description" textfile="$(arg model)"/>
    <param name="use_gui" value="$(arg gui)"/>
    
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
        <param name="use_gui" value="$(arg gui)"/>
        <rosparam param="source_list">["joints_set"]</rosparam>
    </node>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />

    <node name="$(arg s1_node_name)" pkg="openzen_sensor" type="openzen_sensor_node" output="screen" args="_sensor_name:=$(arg s1_name) imu:=$(arg s1_topic_name)"/>
    <node name="$(arg s2_node_name)" pkg="openzen_sensor" type="openzen_sensor_node" output="screen" args="_sensor_name:=$(arg s2_name) imu:=$(arg s2_topic_name)"/>

    <!--launches node-->
    <node name="hak_pub_oz_trac_node" pkg="hak_package" type="hak_pub_oz_trac_node" output="screen" />
</launch>
